e2etest/localserviceimpact_test.go

```diff
@@ -0,0 +1,280 @@
+		Expect(err).ToNot(HaveOccurred())
+		_, err = cluster2.k8s.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: namespace},
+		}, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		_, err = cluster1.k8s.CoreV1().ConfigMaps(namespace).Create(ctx, &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster-info",
+			},
+			Data: map[string]string{
+				"clusterID": "cluster1",
+			},
+		}, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		_, err = cluster2.k8s.CoreV1().ConfigMaps(namespace).Create(ctx, &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster-info",
+			},
+			Data: map[string]string{
+				"clusterID": "cluster2",
+			},
+		}, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		pod := requestPod
+		_, err = cluster1.k8s.CoreV1().Pods(namespace).Create(ctx, &pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		dep := helloDeployment
+		_, err = cluster1.k8s.AppsV1().Deployments(namespace).Create(ctx, &dep, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		_, err = cluster2.k8s.AppsV1().Deployments(namespace).Create(ctx, &dep, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		svc := helloService
+		_, err = cluster1.k8s.CoreV1().Services(namespace).Create(ctx, &svc, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		_, err = cluster2.k8s.CoreV1().Services(namespace).Create(ctx, &svc, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		exp := helloServiceExport
+		_, err = cluster2.mcs.MulticlusterV1alpha1().ServiceExports(namespace).Create(ctx, &exp, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		imp := helloServiceImport
+		_, err = cluster1.mcs.MulticlusterV1alpha1().ServiceImports(namespace).Create(ctx, &imp, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		deploymentAvailable := func(clients clusterClients) func(Gomega) {
+			return func(g Gomega) {
+				deployment, err := clients.k8s.AppsV1().Deployments(namespace).Get(ctx, dep.Name, metav1.GetOptions{})
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(deployment.Status.Conditions).To(ContainElement(Satisfy(func(cond appsv1.DeploymentCondition) bool {
+					return cond.Type == appsv1.DeploymentAvailable &&
+						cond.Status == v1.ConditionTrue
+				})))
+			}
+		}
+		Eventually(deploymentAvailable(cluster1), 30).Should(Succeed())
+		Eventually(deploymentAvailable(cluster2), 30).Should(Succeed())
+		var slices *discoveryv1.EndpointSliceList
+		Eventually(func() int {
+			eps := 0
+			slices, err = cluster2.k8s.DiscoveryV1().EndpointSlices(namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: labels.Set{discoveryv1.LabelServiceName: helloService.Name}.AsSelector().String(),
+			})
+			Expect(err).ToNot(HaveOccurred())
+			for _, s := range slices.Items {
+				eps += len(s.Endpoints)
+			}
+			return eps
+		}, 30).Should(Equal(1))
+		importedSlice := slices.Items[0]
```

Comment:
It might be worth commenting that this is safe because we know there’s at least one slice (thanks to `Should(Equal(1))` just above).